cobigen/core-artifact-retriever/src/test/java/com/devonfw/cobigen/retriever/ArtifactRetrieverTest.java

```diff
@@ -0,0 +1,156 @@
+package com.devonfw.cobigen.retriever;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.devonfw.cobigen.api.exception.CobiGenRuntimeException;
+import com.devonfw.cobigen.retriever.reader.TemplateSetArtifactReader;
+import com.devonfw.cobigen.retriever.settings.MavenSettings;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsModel;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+
+/**
+ * Test class for ArtifactRetriever
+ *
+ */
+public class ArtifactRetrieverTest {
+
+  /**
+   * Test resource location
+   */
+  private final static String testdataRoot = "src/test/resources/testdata/unittest/ArtifactRetrieverTest";
+
+  /**
+   * {@link MavenSettingsModel} without a proxy
+   */
+  private static MavenSettingsModel modelNonProxy;
+
+  /**
+   * {@link MavenSettingsModel} with a proxy
+   */
+  private static MavenSettingsModel modelProxy;
+
+  /**
+   * Maven settings string without a proxy
+   */
+  private static String mavenSettingsNonProxy;
+
+  /**
+   * Maven settings string without a proxy
+   */
+  private static String mavenSettingsProxy;
+
+  /**
+   * WireMock rule to initialize
+   */
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(options().disableRequestJournal());
+
+  /**
+   * Used to initialize data needed for the tests
+   */
+  @BeforeClass
+  public static void setUpClass() {
+
+    try {
+      mavenSettingsNonProxy = new String(Files.readAllBytes(Paths.get(testdataRoot).resolve("settingsNonProxy.xml")));
+      mavenSettingsProxy = new String(Files.readAllBytes(Paths.get(testdataRoot).resolve("settingsProxy.xml")));
+    } catch (IOException e) {
+      throw new CobiGenRuntimeException("Unable to read test settings.xml", e);
```

Comment:
It's a bad idea to throw this exception in a test I think. Maybe use throws instead.

Reply:
Done